crates/router/src/connector/aci/transformers.rs

```diff
@@ -148,176 +353,179 @@ pub enum AciPaymentType {
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for AciPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        let payment_details: PaymentDetails = match item.request.payment_method_data.clone() {
-            api::PaymentMethodData::Card(ccard) => PaymentDetails::AciCard(Box::new(CardDetails {
-                card_number: ccard.card_number,
-                card_holder: ccard.card_holder_name,
-                card_expiry_month: ccard.card_exp_month,
-                card_expiry_year: ccard.card_exp_year,
-                card_cvv: ccard.card_cvc,
-            })),
-            api::PaymentMethodData::PayLater(_) => PaymentDetails::Klarna,
-            api::PaymentMethodData::Wallet(ref wallet_data) => match wallet_data {
-                api_models::payments::WalletData::MbWay(data) => {
-                    PaymentDetails::Wallet(Box::new(WalletPMData {
-                        payment_brand: PaymentBrand::Mbway,
-                        account_id: Some(data.telephone_number.clone()),
-                        shopper_result_url: item.request.router_return_url.clone(),
-                    }))
-                }
-                api_models::payments::WalletData::AliPay { .. } => {
-                    PaymentDetails::Wallet(Box::new(WalletPMData {
-                        payment_brand: PaymentBrand::AliPay,
-                        account_id: None,
-                        shopper_result_url: item.request.router_return_url.clone(),
-                    }))
-                }
+        if item.request.setup_mandate_details.is_some() {
```

Comment:
Why do we still need this as this will be handled in line 375 itself?

Reply:
Removed👍